internal/controllers/machine/machine_controller_status_test.go

```diff
@@ -0,0 +1,1632 @@
+					"name":      "bootstrap-config1",
+					"namespace": metav1.NamespaceDefault,
+				},
+				"status": map[string]interface{}{},
+			}},
+			bootstrapConfigIsNotFound: false,
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineBootstrapConfigReadyV1Beta2Condition,
+				Status: metav1.ConditionTrue,
+				Reason: clusterv1.MachineBootstrapDataSecretProvidedV1Beta2Reason,
+			},
+		},
+		{
+			name:    "mirror Ready condition from bootstrap config",
+			machine: defaultMachine.DeepCopy(),
+			bootstrapConfig: &unstructured.Unstructured{Object: map[string]interface{}{
+				"kind":       "GenericBootstrapConfig",
+				"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
+				"metadata": map[string]interface{}{
+					"name":      "bootstrap-config1",
+					"namespace": metav1.NamespaceDefault,
+				},
+				"status": map[string]interface{}{
+					"conditions": []interface{}{
+						map[string]interface{}{
+							"type":    "Ready",
+							"status":  "False",
+							"message": "some message",
+						},
+					},
+				},
+			}},
+			bootstrapConfigIsNotFound: false,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineBootstrapConfigReadyV1Beta2Condition,
+				Status:  metav1.ConditionFalse,
+				Reason:  clusterv1.MachineBootstrapConfigReadyNoReasonReportedV1Beta2Reason,
+				Message: "some message (from GenericBootstrapConfig)",
+			},
+		},
+		{
+			name:    "Use status.BoostrapReady flag as a fallback Ready condition from bootstrap config is missing",
+			machine: defaultMachine.DeepCopy(),
+			bootstrapConfig: &unstructured.Unstructured{Object: map[string]interface{}{
+				"kind":       "GenericBootstrapConfig",
+				"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
+				"metadata": map[string]interface{}{
+					"name":      "bootstrap-config1",
+					"namespace": metav1.NamespaceDefault,
+				},
+				"status": map[string]interface{}{},
+			}},
+			bootstrapConfigIsNotFound: false,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineBootstrapConfigReadyV1Beta2Condition,
+				Status:  metav1.ConditionFalse,
+				Reason:  clusterv1.MachineBootstrapConfigReadyNoReasonReportedV1Beta2Reason,
+				Message: "GenericBootstrapConfig status.ready is false",
+			},
+		},
+		{
+			name:    "Use status.BoostrapReady flag as a fallback Ready condition from bootstrap config is missing (ready true)",
+			machine: defaultMachine.DeepCopy(),
+			bootstrapConfig: &unstructured.Unstructured{Object: map[string]interface{}{
+				"kind":       "GenericBootstrapConfig",
+				"apiVersion": "bootstrap.cluster.x-k8s.io/v1beta1",
+				"metadata": map[string]interface{}{
+					"name":      "bootstrap-config1",
+					"namespace": metav1.NamespaceDefault,
+				},
+				"status": map[string]interface{}{
+					"ready": true,
+				},
+			}},
+			bootstrapConfigIsNotFound: false,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineBootstrapConfigReadyV1Beta2Condition,
+				Status:  metav1.ConditionFalse,
```

Comment:
Shouldn't this be true? (similar in l.153)

(Btw this is the reason why I wanted to have this additional test case :))